lib/Service/DocumentService.php

```diff
@@ -490,4 +496,41 @@ private function ensureDocumentsFolder(): bool {
 
 		return true;
 	}
+
+	public function lock(int $fileId): bool {
+		if (!$this->lockManager->isLockProviderAvailable()) {
+			return true;
+		}
+
+		$file = $this->getFileById($fileId);
+		try {
+			$this->lockManager->lock(new LockContext(
+				$file,
+				ILock::TYPE_APP,
+				Application::APP_NAME
```

Comment:
Out of topic but this could become `Application::APP_ID` at some point.

Reply:
Yes, something for a separate PR.